lib/dialects/mssql/query-generator.js

```diff
@@ -782,101 +805,38 @@ class MSSQLQueryGenerator extends AbstractQueryGenerator {
-
-    // Handle SQL Server 2008 with TOP instead of LIMIT
-    if (semver.valid(this.sequelize.options.databaseVersion) && semver.lt(this.sequelize.options.databaseVersion, '11.0.0')) {
-      if (options.limit) {
-        topFragment = `TOP ${options.limit} `;
-      }
-      if (options.offset) {
-        const offset = options.offset || 0,
-          isSubQuery = options.hasIncludeWhere || options.hasIncludeRequired || options.hasMultiAssociation;
-        let orders = { mainQueryOrder: [] };
-        if (options.order) {
-          orders = this.getQueryOrders(options, model, isSubQuery);
-        }
-
-        if (!orders.mainQueryOrder.length) {
-          orders.mainQueryOrder.push(this.quoteIdentifier(model.primaryKeyField));
-        }
+  addLimitAndOffset(options) {
+    const composition = new Composition();
 
-        const tmpTable = mainTableAs ? mainTableAs : 'OffsetTable';
-        const whereFragment = where ? ` WHERE ${where}` : '';
-
-        /*
-         * For earlier versions of SQL server, we need to nest several queries
-         * in order to emulate the OFFSET behavior.
-         *
-         * 1. The outermost query selects all items from the inner query block.
-         *    This is due to a limitation in SQL server with the use of computed
-         *    columns (e.g. SELECT ROW_NUMBER()...AS x) in WHERE clauses.
-         * 2. The next query handles the LIMIT and OFFSET behavior by getting
-         *    the TOP N rows of the query where the row number is > OFFSET
-         * 3. The innermost query is the actual set we want information from
-         */
-        const fragment = `SELECT TOP 100 PERCENT ${attributes.join(', ')} FROM ` +
-                        `(SELECT ${topFragment}*` +
-                          ` FROM (SELECT ROW_NUMBER() OVER (ORDER BY ${orders.mainQueryOrder.join(', ')}) as row_num, * ` +
-                            ` FROM ${tables} AS ${tmpTable}${whereFragment})` +
-                          ` AS ${tmpTable} WHERE row_num > ${offset})` +
-                        ` AS ${tmpTable}`;
-        return fragment;
-      }
-      mainFragment = `SELECT ${topFragment}${attributes.join(', ')} FROM ${tables}`;
+    if (options.offset || options.limit) {
+      const offset = options.offset ? new Slot(options.offset) : '0';
+      composition.add('OFFSET ', offset, ' ROWS');
     }
 
-    if (mainTableAs) {
-      mainFragment += ` AS ${mainTableAs}`;
-    }
-
-    if (options.tableHint && TableHints[options.tableHint]) {
-      mainFragment += ` WITH (${TableHints[options.tableHint]})`;
+    if (options.limit) {
+      composition.add(' FETCH NEXT ', new Slot(options.limit), ' ROWS ONLY');
     }
 
-    return mainFragment;
+    return composition;
   }
 
-  addLimitAndOffset(options, model) {
-    // Skip handling of limit and offset as postfixes for older SQL Server versions
-    if (semver.valid(this.sequelize.options.databaseVersion) && semver.lt(this.sequelize.options.databaseVersion, '11.0.0')) {
```

Comment:
Removal of MSSQL <11.0.0 support should probably be well documented in the affected methods.